udata/api/__init__.py

```diff
@@ -100,6 +100,12 @@ def _apply_secure(self, func, permission=None):
 
         @wraps(func)
         def wrapper(*args, **kwargs):
+            if not current_user.is_anonymous and \
+                not current_user.sysadmin and \
+                current_app.config['READ_ONLY_MODE'] and \
+                any(ext in str(func) for ext in current_app.config['METHOD_BLOCKLIST']):
```

Comment:
```suggestion
            if (
                not current_user.is_anonymous and
                not current_user.sysadmin and
                current_app.config['READ_ONLY_MODE'] and 
                any(ext in str(func) for ext in current_app.config['METHOD_BLOCKLIST'])
            ):
```

Personal preference, YMMV